### ISSUE TITLE
container upgrade fails with Error: Mismatched container count post upgrade

### ISSUE THREAD

**Issue by sunilangadi2** [red-hat-storage/cephci#233]
In 4.2 new container got added as ceph-crash due to this while upgrading from 4.x to 4.2
upgrade fails for the task: test_ansible_upgrade - ERROR - Mismatched container count post upgrade
get_container_counts function retuns different number from pre-upgrade and post-upgrade count 
This function should get fixed as per 4.2
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1601970509082/result.html

**Comment by sunilangadi2** [comment]
Pre upgrade container counts: 
```
{'ceph-sangadi-c-u-1601970509082-node8-mds': 1, 
'ceph-sangadi-c-u-1601970509082-node9-rgwnfs': 2, 
'ceph-sangadi-c-u-1601970509082-node11-grafana': 0, 
'ceph-sangadi-c-u-1601970509082-node2-monmgr': 2, 
'ceph-sangadi-c-u-1601970509082-node10-clientnfs': 1, 
'ceph-sangadi-c-u-1601970509082-node3-monrgw': 2, 
'ceph-sangadi-c-u-1601970509082-node7-iscsi-gw': 0, 
'ceph-sangadi-c-u-1601970509082-node6-osdiscsi-gw': 0, 
'ceph-sangadi-c-u-1601970509082-node5-osd': 0, 
'ceph-sangadi-c-u-1601970509082-node4-osd': 0}

```
Post upgrade container counts:
```
{'ceph-sangadi-c-u-1601970509082-node8-mds': 2,
'ceph-sangadi-c-u-1601970509082-node9-rgwnfs': 3,
'ceph-sangadi-c-u-1601970509082-node11-grafana': 0,
'ceph-sangadi-c-u-1601970509082-node2-monmgr': 3,
'ceph-sangadi-c-u-1601970509082-node10-clientnfs': 1,
'ceph-sangadi-c-u-1601970509082-node3-monrgw': 3,
'ceph-sangadi-c-u-1601970509082-node7-iscsi-gw': 0,
'ceph-sangadi-c-u-1601970509082-node6-osdiscsi-gw': 1,
'ceph-sangadi-c-u-1601970509082-node5-osd': 1,
'ceph-sangadi-c-u-1601970509082-node4-osd': 1}

```
**crash container** name not present in client, grafana, nfs, iscsi
It present only in mon,mgr,osd,rgw,mds

But still when we collocate nodes with other nodes container count In-consistent 
It is quite unpredictable. 

I thought of **Not to fail condition** even if container count mismatch
@XDsubhash Please let me know your thoughts on this?

**Comment by XDsubhash** [comment]
for now you can do that .. but raise a issue to fix the compare_container_count function. The issue should have all supported containers daemons list  in rhcs 2.x,3.x,4.x,4.2 ,5.x. It will easy when someone works on it

**Comment by rakeshgm** [comment]
you need to fix `compare_container_count` function. by looking at the above issue, it seems like you are seeing an extra container called ceph-crash which got added into all the nodes. check for extra container and fix the count. this should solve the problem

**Comment by sunilkumarn417** [comment]
@sunilangadi2 I would suggest to consider container counts specific to roles-based(daemons) per each node before upgrade and compare that post upgrade containers count.

**Comment by sunilangadi2** [comment]
@sunilkumarn417 Sure

**Comment by sunilangadi2** [comment]
Fixed this issue in this commit 3fd738ba3318f30bf0b3d3c919dc1ebe213d1fa2
